drush

```diff
@@ -1,131 +1,300 @@
-    php="php"
-  else
-    # Default to using the php that we find on the PATH.
-    # We check for a command line (cli) version of php, and if found use that.
-    # Note that we need the full path to php here for Dreamhost, which behaves oddly.  See http://drupal.org/node/662926
-    php="`which php-cli 2>/dev/null`"
-
-    if [ ! -x "$php" ]; then
-      php="`which php 2>/dev/null`"
-    fi
-
-    if [ ! -x "$php" ]; then
-      echo "ERROR: can't find php."; exit 1
-    fi
-  fi
-fi
-
-# Build the path to drush.php.
-SCRIPT_PATH="`dirname "$SELF_PATH"`/drush.php"
-if [ -n "$CYGWIN" ] ; then
-  # try to determine if we are running cygwin port php or Windows native php:
-  if [ -n "`"$php" -i | grep -E '^System => Windows'`" ]; then
-    SCRIPT_PATH="`cygpath -w -a -- "$SCRIPT_PATH"`"
-  else
-    SCRIPT_PATH="`cygpath -u -a -- "$SCRIPT_PATH"`"
-  fi
-fi
-
-# Check to see if the user has provided a php.ini file or drush.ini file in any conf dir
-# Last found wins, so search in reverse priority order
-for conf_dir in "`dirname "$SELF_PATH"`" /etc/drush "$HOME/.drush" ; do
-  if [ ! -d "$conf_dir" ] ; then
-    continue
-  fi
-  # Handle paths that don't start with a drive letter on MinGW shell. Equivalent to cygpath on Cygwin.
-  if [ -n "$MINGW" ] ; then
-    conf_dir=`sh -c "cd \"$conf_dir\"; pwd -W"`
-  fi
-  if [ -f "$conf_dir/php.ini" ] ; then
-    drush_php_ini="$conf_dir/php.ini"
-  fi
-  if [ -f "$conf_dir/drush.ini" ] ; then
-    drush_php_override="$conf_dir/drush.ini"
-  fi
-done
-# If the PHP_INI environment variable is specified, then tell
-# php to use the php.ini file that it specifies.
-if [ -n "$PHP_INI" ] ; then
-  drush_php_ini="$PHP_INI"
-fi
-# If the DRUSH_INI environment variable is specified, then
-# extract all ini variable assignments from it and convert
-# them into php '-d' options. These will override similarly-named
-# options in the php.ini file
-if [ -n "$DRUSH_INI" ] ; then
-  drush_php_override="$DRUSH_INI"
-fi
-
-# Add in the php file location and/or the php override variables as appropriate
-if [ -n "$drush_php_ini" ] ; then
-  php_options="--php-ini $drush_php_ini"
-fi
-if [ -n "$drush_php_override" ] ; then
-  php_options=`grep '^[a-z_A-Z0-9.]\+ *=' $drush_php_override | sed -e 's|\([^ =]*\) *= *\(.*\)|\1="\2"|' -e 's| ||g' -e 's|^|-d |' | tr '\n\r' '  '`
-fi
-# If the PHP_OPTIONS environment variable is specified, then
-# its contents will be passed to php on the command line as
-# additional options to use.
-if [ -n "$PHP_OPTIONS" ] ; then
-  php_options="$php_options $PHP_OPTIONS"
-fi
-
-# Pass in the path to php so that drush knows which one to use if it
-# re-launches itself to run subcommands.  We will also pass in the php options.
-# Important note: Any options added here must be removed  when Drush processes
-# a #! (shebang) script.  @see drush_adjust_args_if_shebang_script()
-exec "$php" $php_options "$SCRIPT_PATH" --php="$php" --php-options="$php_options" "$@"
+#!/usr/bin/env php
```

Comment:
PHPStorm refuses to recognize this file as PHP as it doesn't understand shebangs. Thats inconvenient but whats worse is that I can't step set breakpoints in non-php files. Could we rename this to drush.finder.php or something and then symlink it as 'drush' in vendor/bin? That would also help people get used to executing the /bin/drush file